flexget/plugins/sites/alpharatio.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import datetime
+import re
+
+from sqlalchemy import Column, Unicode, DateTime
+
+from flexget import plugin, db_schema
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
+from flexget.manager import Session
+from flexget.utils.database import json_synonym
+from flexget.utils.requests import Session as RequestSession
+from flexget.utils.soup import get_soup
+from flexget.config_schema import one_or_more
+from flexget.utils.tools import parse_filesize
+
+log = logging.getLogger('alpharatio')
+Base = db_schema.versioned_base('alpharatio', 0)
+
+requests = RequestSession()
+requests.add_domain_limiter(TimedLimiter('alpharatio.cc', '5 seconds'))
+# ElementZero confirmed with AlphaRato sysop 'jasonmaster' that they do want a 5 second limiter
+
+CATEGORIES = {
+    'tvsd': 'filter_cat[1]',
+    'tvhd': 'filter_cat[2]',
+    'tvdvdrip': 'filter_cat[3]',
+    'tvpacksd': 'filter_cat[4]',
+    'tvpackhd': 'filter_cat[5]',
+    'moviesd': 'filter_cat[6]',
+    'moviehd': 'filter_cat[7]',
+    'moviepacksd': 'filter_cat[8]',
+    'moviepackhd': 'filter_cat[9]',
+    'moviexxx': 'filter_cat[10]',
+    'mvid': 'filter_cat[11]',
+    'gamespc': 'filter_cat[12]',
+    'gamesxbox': 'filter_cat[13]',
+    'gamesps3': 'filter_cat[14]',
+    'gameswii':  'filter_cat[15]',
+    'appspc': 'filter_cat[16]',
+    'appsmac': 'filter_cat[17]',
+    'appslinux': 'filter_cat[18]',
+    'appsmobile': 'filter_cat[19]',
+    '0dayXXX': 'filter_cat[20]',
+    'ebook': 'filter_cat[21]',
+    'audiobook': 'filter_cat[22]',
+    'music': 'filter_cat[23]',
+    'misc': 'filter_cat[24]'
+}
+
+LEECHSTATUS = {
+    'normal': 0,
+    'freeleech': 1,
+    'neutral leech': 2,
+    'either': 3
+}
+
+
+class AlphaRatioCookie(Base):
+    __tablename__ = 'alpharatio_cookie'
+
+    username = Column(Unicode, primary_key=True)
+    _cookie = Column('cookie', Unicode)
+    cookie = json_synonym('_cookie')
+    expires = Column(DateTime)
+
+
+class SearchAlphaRatio(object):
+    """
+        AlphaRatio search plugin.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': one_or_more({'type': 'string', 'enum': list(CATEGORIES.keys())}, unique_items=True),
+            'order_by': {'type': 'string', 'enum': ['seeders', 'leechers', 'time', 'size', 'year', 'snatched'],
+                         'default': 'time'},
+            'order_way': {'type': 'string', 'enum': ['desc', 'asc'], 'default': 'desc'},
```

Comment:
`order_way` is awkward (yes, I realise it's the same in mtv plugin). Maybe change it to `ordering` or `order_desc: yes|no` in both plugins.